### DISCUSSION TITLE
Splash of the white color on page refresh with dark-theme

### DISCUSSION THREAD

**Discussion by dmitriifrlv** [chakra-ui/chakra-ui#5007] [Q&A]
## 📝 Description

I customised a button to have different background colors on dark and light mode. Now, when I turn on  **dark mode** and refresh a page with a theme config set to `initialColorMode: "light"` my button renders first with a background color of light mode, then updates to dark color, and vice versa - with **light mode** on  and `initialColorMode: "dark"` button initially renders  with a dark background and then re-renders and has a correct light color.  

Once the page is loaded, there's no flickering when the theme is toggled. The app is created with create-react-app. 

```
      <Button
        bg={light}
        boxShadow={bsLight}
        _dark={{
          bg: dark,
          boxShadow: bsDark,
        }}
        onClick={toggleColorMode}
        sx={{ width: "10rem" }}
      >
```
      
## :computer: To Reproduce

[Sandbox with reproduced issue](https://codesandbox.io/s/splash-of-white-on-page-refresh-jehmz?file=/src/App.js)

## :notebook_with_decorative_cover: Expected Behavior
1) Page reloads without  color flickering
2) Smooth color transition on theme toggling. 

## :key: Attempts to solve the issue 

I fixed the issue by making theme config dependent on `localStorage`  **"chakra-ui-color-mode"** value : 
```
const config = {
  initialColorMode: localStorage.getItem("chakra-ui-color-mode")
    ? localStorage.getItem("chakra-ui-color-mode")
    : "dark",
  useSystemColorMode: false,
};
```
It fixes the flickering but it feels like a workaround and it's more likely I do something wrong with rewriting my button styles. 
Another possible solution is to set the button transition to `none ` but in this case when the theme is toggled, the background on the button changes faster than on other UI elements and it doesn't provide a user with a nice experience. 

What do I wrong customizing my button and how to avoid color flickering on page re-loads?

**Comment by daniel-schulz-codes** [comment]
Hej @dmitriifrlv we’re currently experiencing an error with Color mode in the current version https://github.com/chakra-ui/chakra-ui/issues/4987 does this match with your experience?

**Reply by dmitriifrlv** [reply]
Hi @takethefake , I do notice that regardless of `config` settings `useSystemColorMode` and `initialColorMode `  the app renders with the **system color mode**. I'm not sure though whether it's the reason that my customized button initially renders according to **light color theme** when **dark mode** is set to `true`  since my system color preference is **dark mode**.

**Reply by daniel-schulz-codes** [reply]
Hej @dmitriifrlv 
Yes this is unfortunately a misbehaviour of the current version. Until we fix this I would suggest to pin the 1.6.10 version Tag of chakra-ui/react

**Reply by dmitriifrlv** [reply]
I've downgraded the library `chakra-ui/react` to ver1.6.10 but the issue is  [still there](https://codesandbox.io/s/splash-of-white-on-page-refresh-jehmz?file=/src/App.js). It seems that I somehow messed up with styles customizing my button but can't figure out what's wrong.
